mock/vcservice.go

```diff
@@ -30,22 +33,24 @@ import (
 // It is used for testing the client which is the coordinator service.
 type VcService struct {
 	protovcservice.ValidationAndCommitServiceServer
-	txBatchChan        chan *protovcservice.Batch
 	numBatchesReceived atomic.Uint32
 	lastCommittedBlock atomic.Pointer[protoblocktx.BlockInfo]
 	txsStatus          *fifoCache[*protoblocktx.StatusWithHeight]
 	txsStatusMu        sync.Mutex
 	healthcheck        *health.Server
 	// MockFaultyNodeDropSize allows mocking a faulty node by dropping some TXs.
 	MockFaultyNodeDropSize int
+	txBatchChannels        map[string]chan *protovcservice.Batch
+	txBatchChannelsMu      sync.Mutex
+	vcIDCounter            atomic.Uint64
```

Comment:
nit: Please align the `vcIDCounter` with the other two to establish a connection between them. For example:
```suggestion
	txBatchChannelsIDCounter atomic.Uint64
```